### ISSUE TITLE
CI should run tests under Python 3.8 (in particular) and others

### ISSUE THREAD

**Issue by webknjaz** [ansible/ansible-navigator#568]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY

The current Zuul CI is misconfigured and doesn't match tox envs:
1) Tox explicitly defines a testenv `py38` expecting that the tests will be run under that interpreter
2) Zuul invokes `tox -e py38` but it gets ignored because it doesn't also pass `--skip-missing-interpreters false`. As a result it **silently** falls back to using Python 3.6:
```console
2021-09-01 12:36:05.448551 \| ubuntu-bionic \| platform linux -- Python 3.6.9, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /home/zuul/src/github.com/ansible/ansible-navigator/.tox/py38/bin/python3
```

This setup is harmful because it creates an impression that the project is being tested under Python 3.8 but in fact, it isn't. We need to improve Zuul setup and probably (additionally) add GHA as the most transparent way of getting all the expected envs to actually run.

**Comment by cidrblock** [comment]
related: 507

**Comment by ssbarnea** [comment]
Closed by #750
